lib/atom-notes.js

```diff
@@ -183,19 +183,67 @@ function handleEvents (self) {
   const {openInterlink} = require('./interlink')
   const {CompositeDisposable, Disposable} = require('atom')
   self.subs = new CompositeDisposable()
-
+  
+  let previewCommand = null
+  
   // user commands
   self.subs.add(
     atom.commands.add('atom-workspace', 'atom-notes:toggle', () => {
       self.getNotesView().toggle()
     }),
-    atom.commands.add('atom-workspace', 'atom-notes:toggle-preview', () => {
-      self.getNotesView().toggle({preview: true})
-    }),
     atom.commands.add('atom-workspace', 'atom-notes:interlink', () => {
       openInterlink()
     })
   )
+  
+  /**
+   * Quickly activate the 'atom-notes:toggle-preview' command if not
+   * already active.
+   */
+  function activatePreviewCommand() {
+    if (!previewCommand) {
+      previewCommand = atom.commands.add(
+        'atom-workspace',
+        'atom-notes:toggle-preview',
+        () => self.getNotesView().toggle({preview: true}))
+      self.subs.add(previewCommand)
+    }
+  }
+  
+  /**
+   * Quickly deactivate the 'atom-notes:toggle-preview' command if is
+   * active.
+   */
+  function deactivatePreviewCommand() {
+    if (previewCommand) {
+      if (self.subs) {
+        self.subs.remove(previewCommand)
+      }
+      previewCommand.dispose()
+      previewCommand = null
+    }
+  }
+  
+  function activatePreview() {
+    if (atom.packages.isPackageActive('markdown-preview-plus') ||
+      atom.packages.isPackageActive('markdown-preview')) {
+      activatePreviewCommand()
+    } else {
+      deactivatePreviewCommand()
+    }
+  }
+  
+  /**
+   * Let's initialize on start
+   */
+  activatePreview()
```

Comment:
So there's a race condition here. It's possible that `atom-notes` is activated before the `markdown-preview*` package(s). To work around this, I would suggest calling `activatePreview()` in a callback for `atom.packages.onDidActivateInitialPackages()` Docs here: https://atom.io/docs/api/v1.26.1/PackageManager#instance-onDidActivateInitialPackages

This would also be a good place to call `atom.packages.onDidActivatePackage(activatePreview)`, so that we aren't triggering that callback until after the initial package activation is done.

It returns a `Disposable` so we'll need to save the return value so that we can properly `.dispose()` the object when we're done with it.